### PR TITLE
Quick eligible atoms fix

### PR DESCRIPTION
**Pull request by pgrinaway**
Two fixes:

* Choose eligible atoms based on torsions that are available (atoms234 have positions)*

* Randomize order of atoms proposed within an "eligible group" and account for this in the logp

*atom1 is by definition always the new atom of interest.

### CODE REVIEW

**Comment by pgrinaway** [comment]
This should address #310

**Comment by pgrinaway** [comment]
The fix I just made passes locally, but I'm waiting for the travis build to start.

File: perses/rjmc/geometry.py [hunk 1]
```diff
@@ -1921,12 +1921,23 @@ def add_atoms(new_atoms, atoms_torsions):
                 The log torsion cchoice probability associated with these added torsions.
 
             """
+            from scipy import special
             logp_torsion_choice = 0.0
             while(len(new_atoms))>0:
                 eligible_atoms = self._atoms_eligible_for_proposal(new_atoms, atoms_with_positions)
+
+                #randomize positions
+                eligible_atoms_in_order = np.random.choice(eligible_atoms, size=len(eligible_atoms), replace=False)
```
Comment (bas-rustenburg) [inline perses/rjmc/geometry.py]: Mostly out of curiosity, any reason not to just shuffle the original array (e.g. using `random.shuffle`)?

File: perses/rjmc/geometry.py [hunk 2]
```diff
@@ -1957,12 +1968,16 @@ def _atoms_eligible_for_proposal(self, new_atoms, atoms_with_positions):
...
+            #get all topological torsions for the appropriate atom
+            torsions = self._get_topological_torsions(atoms_with_positions, atom)
+
+            #go through the topological torsions (atom1 is always the new atom), and if one of them has
+            #atoms 2, 3, 4 in atoms_with_positions, the atom is eligible.
+            for torsion in torsions:
+                if torsion.atom2 in atoms_with_positions and torsion.atom3 in atoms_with_positions and torsion.atom4 in atoms_with_positions:
+                    eligible_atoms.append(atom)
+
+        return list(set(eligible_atoms))
```
Comment (bas-rustenburg) [inline perses/rjmc/geometry.py]: Are parmed.Atom objects hashable? If not, this may result in an error.

File: perses/rjmc/geometry.py (hunk 1 above)
Comment (pgrinaway) [inline perses/rjmc/geometry.py]: I guess I could. I know that by drawing without replacement uniformly, the probability of that configuration will be 1/n!, so I just did that.

File: perses/rjmc/geometry.py (hunk 2 above)
Comment (pgrinaway) [inline perses/rjmc/geometry.py]: Yeah, they are.

Comment (pgrinaway) [inline perses/rjmc/geometry.py]: That line seems to work fine.

**Pull request merged**